main.js

```diff
@@ -273,6 +298,11 @@ app.on('ready', function() {
 
   createWindow();
 
+  if (usingTrayIcon) {
+    const createTrayIcon = require("./app/tray_icon");
+    tray = createTrayIcon(mainWindow, locale.messages);
```

Comment:
I really like how everything has been pulled into a different file! However, there's one thing to consider: there can be multiple values of `mainWindow` over the lifetime of the application. I think, instead of passing `mainWindow` directly here, a function `getMainWindow()` should be passed instead.